core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -72,10 +73,14 @@ public int currentVersion() {
 
   @Override
   public TableMetadata refresh() {
+    boolean currentMetadataWasAvailable = currentMetadata != null;
     try {
       doRefresh();
     } catch (NoSuchTableException e) {
-      LOG.warn("Could not find the table during refresh, setting current metadata to null", e);
+      if (currentMetadataWasAvailable) {
+        LOG.warn("Could not find the table during refresh, setting current metadata to null", e);
+      }
+
       currentMetadata = null;
```

Comment:
I'm not 100% certain that re-setting fields is valid only for when the if statement is true. If then we can move things into if statement, but otherwise I guess it can be left as it is. At least leaving it as it is won't hurt anything.

Reply:
I think this is okay as it is.